lib/internal/modules/helpers.js

```diff
@@ -307,10 +308,58 @@ function lazyLoadTSParser() {
   return parseTS;
 }
 
+function doesTSStripTypesResultMatchSource(code, source) {
+  const sourceBuf = BufferFrom(source);
+  const codeBuf = BufferFrom(code);
+  if (sourceBuf.length !== codeBuf.length) { return false; }
```

Comment:
btw, have you considered doing a comparison of the JS strings? It seems to me it would be much simpler (and maybe more performant) than comparing buffers, especially given the parameters are already strings. wdyt?

```
for (let i = 0; i < source.length; i++) {
  const char = StringPrototypeCharCodePointAt(code, i);
  if (char !== 0x3b && char !== 0x20 && char !== StringPrototypeCharCodePointAt(source, i)) {
     return false;
  }
}
return true;
```

Reply:
@aduh95 will recheck!

Reply:
I didn't use that because the original impl which this rechecks is byte-level based
But it can be done, yes

> and maybe more performant

perf results are inconclusive on this even after initial optimization of this for fast paths -- this is faster for multi-byte but slower for regular ascii sources

will try another version shortly

Reply:
For context, the code I used was:
```js
  function doesTSStripTypesResultMatchSource2(code, source) {
    if (code.length !== source.length) { return false; }
    for (let i = 0; i < code.length; i++) {
      const a = code.codePointAt(i)
      if (a === source.codePointAt(i)) {
        if (a === undefined) break // reached end of both strings
        continue
      }
      if (a !== 0x20 && a !== 0x3b && a !== 0xa0 && a !== 0x2002 && a !== 0xfeff) return false
    }
    return true
  }
```